### PR TITLE
Updated pushingapatch.rst

### PR DESCRIPTION
**Pull request by a-olechtchouk**
Updated pushingapatch.rst for the section "Pushing your changes to the VPP Docs Repository".  Includes the .png files used for screenshots of some steps.

### CODE REVIEW

File: docs/tasks/writingdocs/pushingapatch.rst [hunk 1]
```diff
@@ -1,13 +1,185 @@
...
+Creating a local repository
+---------------------------
+
+Now that you have your own branch of the main repository on Github, you can store it locally on your computer. In your shell, navigate to the directory where you want to store your branch/repo. Then execute:
+
+.. code-block:: shell
+
+   $ git clone https://github.com/YOURUSERNAME/vpp-docs
+
+This will create a directory on your computer named **vpp-docs**, the name of the repo. If you want the directory to be named something other than vpp-docs, you can add your own directory name at the end of the command, shown below:
```
Comment (mdr78) [inline docs/tasks/writingdocs/pushingapatch.rst]: suggest you don't complicate things by describing the option to have the directory called something else?

File: docs/tasks/writingdocs/pushingapatch.rst [hunk 2]
```diff
@@ -1,13 +1,185 @@
...
+
+.. code-block:: shell
+
+   $ git clone https://github.com/YOURUSERNAME/vpp-docs OTHERNAME
+
+
+Now that your branch is on your computer, you can modify and build files however you wish.
+
+
+Building required files
```
Comment (mdr78) [inline docs/tasks/writingdocs/pushingapatch.rst]: Suggest you breakout "building" and "pushing" to readthedocs to a seperate file.

File: docs/tasks/writingdocs/pushingapatch.rst [hunk 3]
```diff
@@ -1,13 +1,185 @@
...
+
+Click "request", and then a dialog will pop-up asking for request approval from fdioDocs. Then click "Request approval from owners."
+
+!!!!!!!!!!TO DO!!!!!!!!!!!!!!! (waiting for request approval).
 
-Initiating a pull request
--------------------------
+
+
+Pushing to your remote branch
```
Comment (mdr78) [inline docs/tasks/writingdocs/pushingapatch.rst]: Suggest you break "pushing" and "merging" to a separate file.

File: docs/tasks/writingdocs/pushingapatch.rst [hunk 4]
```diff
@@ -1,13 +1,185 @@
...
+   $ git clone https://github.com/YOURUSERNAME/vpp-docs OTHERNAME
+
+
+Now that your branch is on your computer, you can modify and build files however you wish.
+
+
+Building required files
+-----------------------
+
+Since we use **.rst** files, **Sphinx**, and **Read the docs** for documenting, you should install `Sphinx <http://www.sphinx-doc.org/en/master/usage/installation.html>`_, and follow their `getting started guide <http://www.sphinx-doc.org/en/master/usage/quickstart.html>`_.
```
Comment (mdr78) [inline docs/tasks/writingdocs/pushingapatch.rst]: FD.io VPP Documentation uses reStructuredText (rst)

File: docs/tasks/writingdocs/pushingapatch.rst [hunk 5]
```diff
@@ -1,13 +1,185 @@
...
+
+Building these files will generate an **index.html** file, which you can then view in your browser to verify and see your file changes.
+
+Make sure your Sphinx-build directory is **not** the directory of your local repo, or else Git will think you want to stage and commit your build files (which will show up as "Untracked files" when you try to commit). To ensure this, create a new empty directory which will store your build files.
```
Comment (mdr78) [inline docs/tasks/writingdocs/pushingapatch.rst]: ok you don't solve this problem with documentation, you solve this problem by creating a gitignore file in the repo (https://git-scm.com/docs/gitignore), see https://git.fd.io/vpp/tree/.gitignore for an example.

**Comment by a-olechtchouk** [comment]
Incorporated Ray and John's comments (Moving the branch section to a new page, removing console output pictures into code blocks, and adding additional git commands). 

You can view the docs here [https://a-olechtchoukvpp-docs.readthedocs.io/en/latest/tasks/writingdocs/index.html](https://a-olechtchoukvpp-docs.readthedocs.io/en/latest/tasks/writingdocs/index.html).

**Pull request merged**